File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> prefixCount({{0,1}});
        int prefix=0, count=0;
        for(int num: nums){
            prefix+=num;
            if(prefixCount.find(prefix-k)!=prefixCount.end())
                count+=prefixCount[prefix-k];
            prefixCount[prefix]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        running_sum = 0
        prefixSumCount = defaultdict(int, {0: 1})
        for i in range(len(nums)):
            running_sum += nums[i]
            if (running_sum - k) in prefixSumCount:
                count += prefixSumCount[running_sum-k]
            prefixSumCount[running_sum]+=1
        return count